kitsune/sumo/middleware.py

```diff
@@ -104,6 +104,26 @@ def process_response(self, request, response):
         return response
 
 
+class VaryNoCacheMiddleware(MiddlewareMixin):
+    """
+    If enabled this will set headers to prevent the CDN (or other caches) from
+    caching the response if the response was set to vary on accept-langauge.
+    This should be near the top of the list of middlewares so it will be able
+    to inspect the near-final response since response middleware is processed
+    in reverse.
+    """
+    def __init__(self):
+        if not settings.ENABLE_VARY_NOCACHE_MIDDLEWARE:
+            raise MiddlewareNotUsed
+
+    @staticmethod
+    def process_response(request, response):
+        if 'vary' in response and 'accept-language' in response['vary'].lower():
```

Comment:
Does the response object *always* have a 'vary' key, or should this be `response.get('vary', '')`